### PR TITLE
Dryrun mechanisms

### PR DESCRIPTION
**Pull request by fouriaux**
for getting a better load balancing, we need more accurate measurement of mechanisms execution in coreneuron. Previous version was using estimation from Neuron, however we have experienced some different figures when directly estimating time in coreneuron. 
This version is executing each mechanism in coreneuron instead of relying on neuron estimation.

### CODE REVIEW

File: coreneuron/nrniv/dryrun.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ *  Dryrun mode of coreneuron, computing execution statistics and estimators 
+ *  without a full simulation execution
+ */
+
+#ifndef dryrun_h
+#define dryrun_h
+
+extern "C"
```
Comment (ohm314) [inline coreneuron/nrniv/dryrun.h]: why is this needed?

**Comment by pramodk** [comment]
dry-run mode will be implemented later after planned refactoring of the code base.
